src/components/Message/Message.js

```diff
@@ -1,366 +1,300 @@
-      this.state = {
-        loading: false,
-      };
-    }
-
-    static propTypes = {
-      /**
-       * Attachment UI component to display attachment in individual message.
-       * Available from [channel context](https://getstream.github.io/stream-chat-react-native/#channelcontext)
-       * */
-      Attachment: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
-      /** The current channel this message is displayed in */
-      channel: PropTypes.object.isRequired,
-      /** The client connection object for connecting to Stream */
-      client: PropTypes.object.isRequired,
-      /** Disables the message UI. Which means, message actions, reactions won't work. */
-      disabled: PropTypes.bool,
-      /** @see See [Keyboard Context](https://getstream.github.io/stream-chat-react-native/#keyboardcontext) */
-      dismissKeyboard: PropTypes.func,
-      /** Should keyboard be dismissed when messaged is touched */
-      dismissKeyboardOnMessageTouch: PropTypes.bool,
-      /** Editing, if the message is currently being edited */
-      editing: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
-      /** groupStyles, a list of styles to apply to this message. ie. top, bottom, single etc */
-      groupStyles: PropTypes.array,
-      /** Latest message id on current channel */
-      lastReceivedId: PropTypes.string,
-      /** The message object */
-      message: PropTypes.object.isRequired,
-      /**
-       * Message UI component to display a message in message list.
-       * Available from [channel context](https://getstream.github.io/stream-chat-react-native/#channelcontext)
-       * */
-      Message: PropTypes.oneOfType([PropTypes.node, PropTypes.elementType]),
+const MessageWithContext = React.memo((props) => {
+  const {
+    channel,
+    dismissKeyboard,
+    dismissKeyboardOnMessageTouch = true,
+    emojiData,
+    message,
+    Message,
+    onMessageTouch: onMessageTouchProp,
+    removeMessage,
+    retrySendMessage,
+    setEditingState,
+    updateMessage,
+  } = props;
+
+  const { client } = useContext(ChatContext);
+
+  const isMyMessage = (message) => client.user.id === message.user.id;
+
+  const isAdmin = () =>
+    client.user.role === 'admin' ||
+    (channel.state &&
+      channel.state.membership &&
+      channel.state.membership.role === 'admin');
+
+  const isOwner = () =>
+    channel.state &&
+    channel.state.membership &&
+    channel.state.membership.role === 'owner';
+
+  const isModerator = () =>
+    channel.state &&
+    channel.state.membership &&
+    (channel.state.membership.role === 'channel_moderator' ||
+      channel.state.membership.role === 'moderator');
+
+  const canEditMessage = () =>
+    isMyMessage(message) || isModerator() || isOwner() || isAdmin();
+
+  const canDeleteMessage = () => canEditMessage();
+
+  const handleFlag = async () => await client.flagMessage(message.id);
+
+  const handleMute = async () => await client.muteUser(message.user.id);
```

Comment:
do these need to be async await cause you're returning them already

Reply:
these aren't implemented anywhere yet so they're basically just placeholders, not sure if the idea is to add more logic to these functions as soon as we add them to the SDK